doc/start/hardware-recommendations.rst

```diff
@@ -236,6 +236,48 @@ costs.  Some RAID HBAs can be configured with an IT-mode "personality".
    Throughput 2`_ for additional details.
 
 
+Benchmarking
+------------
+
+BlueStore opens block devices in O_DIRECT and uses fsync frequently to
+ensure that data is safely persisted to media. You can evaluate a drive's
+low-level write performance using ``fio``. For example, 4kB random write
+performance is measured as follows:
+
+   ``fio --name=/dev/sdX --ioengine=libaio --direct=1 --fsync=1 --readwrite=randwrite --blocksize=4k --runtime=300``
+
+
+Disk Caches
+-----------
+
+Most enterprise-grade SSDs and HDDs include power loss protection features
+which employ multi-level caches to speed up direct random writes. Many of
+these types of devices can be toggled between two caching modes -- a
+volatile cache flushed safely with fsync, or a non-volatile cache written
+synchronously. Counterintuitively, these two modes are selected by either
+"enabling" or "disabling" the device cache.
+
+The cache configuration can be queried with ``hdparm -W`` or by reading the
+values in ``/sys/devices/scsi_disk/*/cache_type``. The default configuration
+-- normally cache enabled / "write back" -- is often not optimal. The
+alternative -- cache disabled / "write through" -- is often observed to
+dramatically increase random write performance on both modern SSDs and HDDs.
+
+Users are therefore encouraged to benchmark new devices with ``fio`` as
+described earlier and persist the optimal cache configuration for their
+devices.
+
+.. tip:: Modifying cache_type sets the device cache status; on some distros
```

Comment:
Which distros?  When should a user do this vs the udev rule below?  This should be fleshed out a little more.

Reply:
I've fleshed it out, better now?